tron/core/actionrun.py

```diff
@@ -590,7 +590,9 @@ def fail(self, exit_status=None):
 
         return self._done("fail", exit_status)
 
-    def _exit_unsuccessful(self, exit_status=None, retry_original_command=True) -> Optional[Union[bool, ActionCommand]]:
+    def _exit_unsuccessful(
+        self, exit_status=None, retry_original_command=True, non_retryable_exit_codes=()
```

Comment:
```suggestion
        self, exit_status=None, retry_original_command=True, non_retryable_exit_codes=[]
```

Since we're validating that non_retryable_exit_codes is a list w/ build_list_of_type... in our validator, we should probably set the default here to empty list instead of empty tuple? In practice, would probably have no effect but would help future us when debugging and prevent us wondering when we turned this into a tuple :p

Reply:
I did initially a list but changed into a tuple since build_list_of_type returns tuple

https://github.com/Yelp/Tron/blob/b8e91c32c8f78e7b6788da98cd7f79e1d3b098c7/tron/config/config_utils.py#L193

Reply:
Hrm, that **is** confusing since we also use `build_list_of_type_validator` for e.g. `secret_volumes`, and then type that as a `List` in https://github.com/Yelp/Tron/blob/master/tron/core/action.py#L36 !

Reply:
OK, reading the build_list_of_type cmd again: the `validator()` will return a tuple of validators, but the actual _values_ it is validating should be lists, and it validates it as such here: https://github.com/Yelp/Tron/blob/b8e91c32c8f78e7b6788da98cd7f79e1d3b098c7/tron/config/config_utils.py#L189C15-L189C25 .  

Therefore, our default values for these config options should be lists, as I understand it.